lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1025,26 +1040,409 @@ spec = do
+                        ]
+                    , assertions =
+                        [ expectListSizeEqual 0 ]
+                    }
+                , TestCase -- 7
+                    { query = toQueryString
+                        [ ("start", utcIso8601ToText $ plusOneSecond t1)
+                        , ("order", "ascending")
+                        ]
+                    , assertions =
+                        [ expectListSizeEqual 1
+                        , expectListItemFieldEqual 0 amount a2
+                        ]
+                    }
+                , TestCase -- 8
+                    { query = toQueryString
+                        [ ("order", "descending")
+                        , ("start", utcIso8601ToText $ plusOneSecond t1)
+                        , ("end", utcIso8601ToText t2)
+                        ]
+                    , assertions =
+                        [ expectListSizeEqual 1
+                        , expectListItemFieldEqual 0 amount a2
+                        ]
+                    }
+                , TestCase -- 9
+                    { query = toQueryString
+                        [ ("order", "ascending")
+                        , ("start", utcIso8601ToText $ minusOneSecond t1)
+                        , ("end", utcIso8601ToText $ minusOneSecond t2)
+                        ]
+                    , assertions =
+                        [ expectListSizeEqual 1
+                        , expectListItemFieldEqual 0 amount a1
+                        ]
+                    }
+                , TestCase -- 10
+                    { query = toQueryString
+                        [ ("order", "descending")
+                        , ("start", utcIso8601ToText $ minusOneSecond t1)
+                        ]
+                    , assertions =
+                        [ expectListSizeEqual 2
+                        , expectListItemFieldEqual 0 amount a2
+                        , expectListItemFieldEqual 1 amount a1
+                        ]
+                    }
+                , TestCase -- 11
+                    { query = toQueryString
+                        [ ("start", utcIso8601ToText $ minusOneSecond t1)
+                        , ("end", utcIso8601ToText t2)
+                        ]
+                    , assertions =
+                        [ expectListSizeEqual 2
+                        , expectListItemFieldEqual 0 amount a2
+                        , expectListItemFieldEqual 1 amount a1
+                        ]
+                    }
+                , TestCase -- 12
+                    { query = toQueryString
+                        [ ("start", utcIso8601ToText $ minusOneSecond t1)
+                        , ("end", utcIso8601ToText $ plusOneSecond t2)
+                        ]
+                    , assertions =
+                        [ expectListSizeEqual 2
+                        , expectListItemFieldEqual 0 amount a2
```

Comment:
I was curious if it was possible to remove `expectListItemFieldEqual 0 amount` in favour for like `expectFieldEqual (0_ . amount)`, but I didn't figure out how with generic-lens. Arguably that would be less readable also.